### ISSUE TITLE
Use listenOnce in amp-story

### ISSUE THREAD

**Issue by newmuis** [ampproject/amphtml#13159]
We have many places in `amp-story` code that add an event listener but do not remove it.  For this, we can refactor to use [`listenOnce`](https://github.com/ampproject/amphtml/blob/master/src/event-helper.js#L78).

**Comment by newmuis** [comment]
This is a code cleanup that can happen during a fixit.

**Comment by juanlizarazo** [comment]
I can work on this one

**Comment by newmuis** [comment]
Sure!  Thanks @juanlizarazo!

**Comment by juanlizarazo** [comment]
@newmuis I attempted to use `listenOnce` but that one removes the listener as soon as the event is received, it just executes each handler once for each event, e.g. next page would work just once.

So instead, I refactored this with [listen](https://github.com/ampproject/amphtml/blob/master/src/event-helper.js#L55) and stored the unlisteners than then are called in unlayoutCallback.

PR #17004 is ready for review.
